### PR TITLE
Quac/feat/add unit tests

### PR DESCRIPTION
**Pull request by quac88**
## Description of what you changed
Created unit tests, fixed error handling in several methods, fixed edge case handling in several methods

## Did you add any new dependencies ex (using a new library, python version, etc)?
N/A

## How did you test your changes?
N/A

## What issue (if any) does this fix?
Added unit testing. Fixed minor edge cases that likely wouldn't have been an issue anyways

## Checklist:
- [x] I have read the [contributing.md](contributing.md) rules
- [x] I am follwoing the [branch naming rules](contributing.md#format-for-branch-names)
- [x] I am using symantic release and angular style commits (see [commits.md](docs/commits.md))
- [x] I have tested my changes
- [x] I have added any new dependencies to the [README.md](README.md)
- [x] I have added any new dependencies to the [requirements.txt](requirements.txt)

### CODE REVIEW

File: src/poker/game.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from __future__ import annotations
 import random
-from cardecky import Deck, HandRanker
+from src.poker.cardecky import Deck, HandRanker
```
Comment (MichaelAGolden) [inline src/poker/game.py]: Have you been able to run this without using src.poker as a prepend to the cardecky.py file? I'm not quite clear why it gives me issues but none of the files work for me at all using this syntax

Comment (quac88) [reply src/poker/game.py]: you're right, fixing it now

**Review by quac88 — COMMENTED** [review]
LGTM
